From bb36794b6bedc86ad400d7b58f00bd56734d37b4 Mon Sep 17 00:00:00 2001
From: Rune Lillesveen <futhark@chromium.org>
Date: Mon, 27 May 2024 11:54:17 +0000
Subject: [PATCH] Fix nullptr RuleSet crash

Use the RuleSet from active sheets instead of retrieving it from the
StyleSheetContents.

Bug: 339546252
Change-Id: I65f7d78049a504d8aaa1499acd3f053a867a9737
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5563645
Reviewed-by: Anders Hartvoll Ruud <andruud@chromium.org>
Commit-Queue: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1306368}
---

--- a/third_party/blink/renderer/core/css/resolver/scoped_style_resolver.cc
+++ b/third_party/blink/renderer/core/css/resolver/scoped_style_resolver.cc
@@ -141,7 +141,7 @@
     StyleSheetContents* contents = sheet->Contents();
     if (contents->HasOneClient() ||
         visited_shared_style_sheet_contents.insert(contents).is_new_entry) {
-      features.Merge(contents->GetRuleSet().Features());
+      features.Merge(rule_set->Features());
     }
   }
 }
